internal/initializer/tls.go

```diff
@@ -0,0 +1,341 @@
+/*
+Copyright 2023 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package initializer
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+)
+
+const (
+	errGenerateCA              = "cannot generate CA certificate"
+	errParseCACertificate      = "cannot parse CA certificate"
+	errParseCAKey              = "cannot parse CA key"
+	errLoadOrGenerateSigner    = "cannot load or generate certificate signer"
+	errDecodeKey               = "cannot decode key"
+	errDecodeCert              = "cannot decode cert"
+	errFmtGetTLSSecret         = "cannot get TLS secret: %s"
+	errFmtCannotCreateOrUpdate = "cannot create or update secret: %s"
+)
+
+const (
+	// RootCACertSecretName is the name of the secret that will store CA certificates and rest of the
+	// certificates created per entities will be signed by this CA
+	RootCACertSecretName = "xp-root-ca"
```

Comment:
@ezgidemirel can you switch it to `crossplane-root-ca` as agreed?

Reply:
done